File: src/main/java/carelog/carelog/relation/app/RelationService.java
package carelog.carelog.relation.app;

import carelog.carelog.auth.app.CustomUserDetails;
import carelog.carelog.relation.domain.*;
import carelog.carelog.relation.web.dto.*;

import java.util.*;

public interface RelationService {

    RelationResponse createRelation(UUID customerPublicId, CustomUserDetails userDetails);

    RelationResponse findRelationByPublicId(UUID relationPublicId, CustomUserDetails userDetails);

    RelationResponse findRelationByManagerAndCustomer(UUID managerPublicId, UUID customerPublicId, CustomUserDetails userDetails);

    List<RelationResponse> findAllRelationsByManager(CustomUserDetails userDetails);

    List<RelationResponse> findAllRelationsByCustomer(UUID customerPublicId, CustomUserDetails userDetails);

    RelationResponse updateRelationsStatus(UUID relationPublicId, RelationStatus newStatus, CustomUserDetails userDetails);

    void deleteRelation(UUID relationPublicId, CustomUserDetails userDetails);
}

File: src/main/kotlin/carelog/carelog/relation/app/RelationService.kt
package carelog.carelog.relation.app

import carelog.carelog.auth.app.CustomUserDetails
import carelog.carelog.relation.domain.RelationStatus
import carelog.carelog.relation.web.dto.RelationResponse
import java.util.UUID

interface RelationService {
    fun createRelation(customerPublicId: UUID, userDetails: CustomUserDetails): RelationResponse
    fun findRelationByPublicId(relationPublicId: UUID, userDetails: CustomUserDetails): RelationResponse
    fun findRelationByManagerAndCustomer(managerPublicId: UUID, customerPublicId: UUID, userDetails: CustomUserDetails): RelationResponse
    fun findAllRelationsByManager(userDetails: CustomUserDetails): List<RelationResponse>
    fun findAllRelationsByCustomer(customerPublicId: UUID, userDetails: CustomUserDetails): List<RelationResponse>
    fun updateRelationsStatus(relationPublicId: UUID, newStatus: RelationStatus, userDetails: CustomUserDetails): RelationResponse
    fun deleteRelation(relationPublicId: UUID, userDetails: CustomUserDetails)
}
